### PR TITLE
feat(ci): standardized per-platform smoke-sketch manifest (refs #2411)

### PR DESCRIPTION
**Pull request by zackees**
## Summary

Implements **Part A** of #2411: a canonical, platform-agnostic list of FastLED example sketches that EVERY supported board must compile in CI. This is the prerequisite the opt-in PlatformIO comparison job (Part B) will read from.

Files added:

- `tests/platforms/_standard/SMOKE_SKETCHES.txt` — the manifest itself (plain text, one example name per line, `#` comments, blank lines ignored).
- `ci/standardized_smoke_sketches.py` — a tiny reader (`load_smoke_sketches()`, `load_smoke_sketches_detailed()`, `manifest_path()`) plus a CLI entry point that prints the list, suitable for shell pipelines.
- `ci/tests/test_standardized_smoke_sketches.py` — 8 tests covering parser edge cases plus three live-manifest sanity checks (manifest exists, contains the canonical four, every entry resolves to an existing `examples/<name>/` directory).

## Canonical sketches in the manifest

| Sketch | Why it's in the set |
|---|---|
| `Blink` | Basic clockless one-pixel sanity. If this fails, the toolchain can't link the simplest FastLED program. |
| `XYMatrix` | 2D matrix wiring + `XYMap` — the most common 2D abstraction. |
| `Apa102` | Clocked SPI chipset + 8-bit gamma / palette code. Picked over `Apa102HD` because `Apa102` has the smaller exclusion set (only `stm32f103cb` is filtered out via `@filter`) and matches the "PROGMEM gradient/palette tables" intent in the issue. |
| `AudioFftParity` | FFT / sin LUT path. ESP32-only by its own `@filter` (`esp_dsp.h` availability — see #2623); on non-ESP32 boards the existing `sketch_filter` machinery skips it cleanly. The slot is reserved here so a future cross-platform FFT-parity sketch can replace this entry without churning every CI consumer. |

Per-board skipping continues to be driven by the existing `@filter` directives inside each `.ino` — the manifest itself stays platform-agnostic on purpose. Per-board overrides / additional smoke sketches remain under `tests/platforms/<board>/` as today.

## Sample invocation

```bash
$ uv run python ci/standardized_smoke_sketches.py
Blink
XYMatrix
Apa102
AudioFftParity

# Shell-pipeline consumer (illustrative):
$ bash compile esp32dev --examples $(uv run python ci/standardized_smoke_sketches.py | tr '\n' ' ')
```

Python consumers:

```python
from ci.standardized_smoke_sketches import load_smoke_sketches
for sketch in load_smoke_sketches():
    ...  # feed to ci-compile.py / fbuild / pio_compile / arduino-cli
```

## Not in scope (deferred)

- **Part B from #2411** — the opt-in `also_build_with_platformio` input on `.github/workflows/build_template.yml` and the per-backend pass/fail summary. This PR is the prerequisite; the workflow side lands in a follow-up.
- **CI matrix shape** — no changes to the default per-PR build matrix. Existing per-board build runs are unchanged.
- **Part C (agent docs update)** — also deferred to the Part B follow-up so the "diagnosing fbuild-only failures" docs section can reference both pieces.

## Test plan

- [x] `uv run python -m pytest ci/tests/test_standardized_smoke_sketches.py -v` — 8/8 pass locally (covers parser comments / blank lines / inline comments / duplicate rejection / missing-manifest error, plus live-manifest sanity).
- [x] `bash lint` — clean (ruff + ty + cpp_linting + javascript_linting + meson_linting).
- [x] `uv run python ci/standardized_smoke_sketches.py` — prints the four canonical names, one per line.
- [ ] CI: GitHub Actions on this PR.

Refs #2411.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **Tests**
  * Added comprehensive test coverage validating the canonical smoke-sketch manifest, that entries resolve to example sketches, and parser behaviors (skips blanks/comments, strips inline comments, records line numbers, rejects duplicates, errors on missing manifest).

* **Chores**
  * Introduced a standardized smoke-sketch manifest for CI across platforms and a small CLI to list the canonical sketches.

### CODE REVIEW

File: ci/standardized_smoke_sketches.py [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+
+    Attributes:
+        name: Example name as it appears in the manifest (e.g. ``"Blink"``).
+              Resolves to ``examples/<name>/`` under the project root.
+        manifest_line: 1-indexed line number in the manifest the entry came
+              from. Useful for error messages.
+    """
+
+    name: str
+    manifest_line: int
```
[the remark this answers is not in the document]
Comment (zackees) [reply ci/standardized_smoke_sketches.py:52]: Addressed in commit 20b027ade: added `from typeguard import typechecked` and `@typechecked` above `@dataclass(frozen=True)` on `SmokeSketch`.

File: ci/standardized_smoke_sketches.py [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+        project_root: Optional override (used by tests). Defaults to the
+            FastLED repo root inferred from this file's location.
+    """
+    root = project_root if project_root is not None else _project_root()
+    return root / MANIFEST_RELATIVE_PATH
+
+
+def load_smoke_sketches_detailed(
+    project_root: Path | None = None,
+) -> list[SmokeSketch]:
```
[the remark this answers is not in the document]
Comment (zackees) [reply ci/standardized_smoke_sketches.py]: Addressed in commit 20b027ade: removed `= None` defaults from `manifest_path`, `load_smoke_sketches_detailed`, and `load_smoke_sketches`. Callers in `_main` and `ci/tests/test_standardized_smoke_sketches.py` now pass `None` explicitly to mean 'use `_project_root()`'.

File: ci/standardized_smoke_sketches.py [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+def load_smoke_sketches(project_root: Path | None = None) -> list[str]:
+    """Return just the sketch names from the manifest, in manifest order.
+
+    Convenience wrapper around :func:`load_smoke_sketches_detailed` for
+    callers (e.g. shell pipelines) that don't need line-number metadata.
+
+    Args:
+        project_root: Optional override (used by tests).
+    """
+    return [entry.name for entry in load_smoke_sketches_detailed(project_root)]
```
[the remark this answers is not in the document]
Comment (zackees) [reply ci/standardized_smoke_sketches.py]: Addressed in commit 20b027ade: replaced the list comprehension in `load_smoke_sketches` with an explicit loop that appends to a typed `names: list[str]` and returns it.

File: ci/standardized_smoke_sketches.py [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+    """
+    try:
+        for name in load_smoke_sketches():
+            print(name)
+    except FileNotFoundError as exc:
+        print(f"ERROR: {exc}")
+        return 1
+    except ValueError as exc:
+        print(f"ERROR: {exc}")
+        return 2
```
[the remark this answers is not in the document]
Comment (zackees) [reply ci/standardized_smoke_sketches.py:153]: Addressed in commit 20b027ade: `_main` now catches `KeyboardInterrupt` first and routes it through the project-wide `handle_keyboard_interrupt(ki)` helper (required by the KBI002 lint rule — bare `raise` is rejected). Returns 130 on Ctrl-C; `FileNotFoundError` / `ValueError` handlers come after.

File: ci/standardized_smoke_sketches.py [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+    return 0
+
+
+if __name__ == "__main__":
+    import sys
+
```
[the remark this answers is not in the document]
Comment (zackees) [reply ci/standardized_smoke_sketches.py]: Addressed in commit 20b027ade: `import sys` is now in the top-level stdlib import block; removed the local import inside the `if __name__ == "__main__":` guard.
